### ISSUE TITLE
Random crash in MP

### ISSUE THREAD

**Issue by twothe** [ricksouth/serilum-mc-mods#2012]
## **Information**
Minecraft version: 1.20.1
Modloader: Forge 
Forge version: 47.2.1
Environment: Multiplayer

Mod name: No Hostiles Around Campfires
Mod version: 5.6

## **Description**
The server crashed randomly when a mob spawn was attempted to be prevented.
There might be an interaction with Apotheosis.

## **Crash report**
```
[27Oct2023 20:32:23.379] [Server thread/ERROR] [net.minecraftforge.eventbus.EventBus/EVENTBUS]: Exception caught during firing event: Late invocations of Mob#setSpawnCancelled are not permitted.
        Index: 8
        Listeners:
                0: HIGHEST
                1: ASM: com.natamus.collective.forge.events.RegisterCollectiveEvents@721362bb onMobSpawnerSpecialSpawn(Lnet/minecraftforge/event/entity/living/MobSpawnEvent$FinalizeSpawn;)V
                2: HIGH
                3: ASM: class dev.architectury.event.forge.EventHandlerImplCommon eventLivingSpawnEvent(Lnet/minecraftforge/event/entity/living/MobSpawnEvent$FinalizeSpawn;)V
                4: NORMAL
                5: net.minecraftforge.eventbus.EventBus$$Lambda$1464/0x000000080166d380@7dbee981
                6: ASM: class com.minecolonies.coremod.event.EventHandler on(Lnet/minecraftforge/event/entity/living/MobSpawnEvent$FinalizeSpawn;)V
                7: ASM: class com.hollingsworth.arsnouveau.common.event.EventHandler livingSpawnEvent(Lnet/minecraftforge/event/entity/living/MobSpawnEvent$FinalizeSpawn;)V
                8: ASM: com.natamus.nohostilesaroundcampfire.forge.events.ForgeCampfireEvent@4600cf32 onEntityCheckSpawn(Lnet/minecraftforge/event/entity/living/MobSpawnEvent$FinalizeSpawn;)V
                9: LOW
                10: ASM: dev.shadowsoffire.apotheosis.adventure.AdventureEvents@159958 special(Lnet/minecraftforge/event/entity/living/MobSpawnEvent$FinalizeSpawn;)V
                11: ASM: dev.shadowsoffire.apotheosis.adventure.boss.BossEvents@1db0ef84 minibosses(Lnet/minecraftforge/event/entity/living/MobSpawnEvent$FinalizeSpawn;)V
                12: ASM: dev.shadowsoffire.apotheosis.adventure.boss.BossEvents@1db0ef84 naturalBosses(Lnet/minecraftforge/event/entity/living/MobSpawnEvent$FinalizeSpawn;)V
                13: LOWEST
                14: ASM: mcjty.incontrol.ForgeEventHandlers@22ddac17 onEntitySpawnEvent(Lnet/minecraftforge/event/entity/living/MobSpawnEvent$FinalizeSpawn;)V
java.lang.UnsupportedOperationException: Late invocations of Mob#setSpawnCancelled are not permitted.
        at TRANSFORMER/minecraft@1.20.1/net.minecraft.world.entity.Mob.setSpawnCancelled(Mob.java:1530)
        at TRANSFORMER/forge@47.2.1/net.minecraftforge.event.entity.living.MobSpawnEvent$FinalizeSpawn.setSpawnCancelled(MobSpawnEvent.java:389)
        at TRANSFORMER/nohostilesaroundcampfire@5.6/com.natamus.nohostilesaroundcampfire.forge.events.ForgeCampfireEvent.onEntityCheckSpawn(ForgeCampfireEvent.java:44)
        at TRANSFORMER/nohostilesaroundcampfire@5.6/com.natamus.nohostilesaroundcampfire.forge.events.__ForgeCampfireEvent_onEntityCheckSpawn_FinalizeSpawn.invoke(.dynamic)
        at MC-BOOTSTRAP/net.minecraftforge.eventbus/net.minecraftforge.eventbus.ASMEventHandler.invoke(ASMEventHandler.java:73)
        at MC-BOOTSTRAP/net.minecraftforge.eventbus/net.minecraftforge.eventbus.EventBus.post(EventBus.java:315)
        at MC-BOOTSTRAP/net.minecraftforge.eventbus/net.minecraftforge.eventbus.EventBus.post(EventBus.java:296)
        at TRANSFORMER/forge@47.2.1/net.minecraftforge.event.ForgeEventFactory.onFinalizeSpawn(ForgeEventFactory.java:281)
        at TRANSFORMER/minecraft@1.20.1/net.minecraft.world.entity.monster.Zombie.m_214076_(Zombie.java:391)
        at TRANSFORMER/minecraft@1.20.1/net.minecraft.world.entity.LivingEntity.m_6667_(LivingEntity.java:1325)
        at TRANSFORMER/minecraft@1.20.1/net.minecraft.world.entity.npc.AbstractVillager.m_6667_(AbstractVillager.java:178)
        at TRANSFORMER/minecraft@1.20.1/net.minecraft.world.entity.npc.Villager.m_6667_(Villager.java:570)
        at TRANSFORMER/minecraft@1.20.1/net.minecraft.world.entity.LivingEntity.m_6469_(LivingEntity.java:1182)
        at TRANSFORMER/minecraft@1.20.1/net.minecraft.world.entity.Mob.m_7327_(Mob.java:1410)
        at TRANSFORMER/minecraft@1.20.1/net.minecraft.world.entity.monster.Zombie.m_7327_(Zombie.java:315)
        at TRANSFORMER/minecraft@1.20.1/net.minecraft.world.entity.monster.Husk.m_7327_(Husk.java:57)
        at TRANSFORMER/minecraft@1.20.1/net.minecraft.world.entity.ai.goal.MeleeAttackGoal.m_6739_(MeleeAttackGoal.java:147)
        at TRANSFORMER/minecraft@1.20.1/net.minecraft.world.entity.ai.goal.MeleeAttackGoal.m_8037_(MeleeAttackGoal.java:138)
        at TRANSFORMER/minecraft@1.20.1/net.minecraft.world.entity.ai.goal.ZombieAttackGoal.m_8037_(ZombieAttackGoal.java:28)
        at TRANSFORMER/minecraft@1.20.1/net.minecraft.world.entity.ai.goal.WrappedGoal.m_8037_(WrappedGoal.java:65)
        at TRANSFORMER/minecraft@1.20.1/net.minecraft.world.entity.ai.goal.GoalSelector.m_186081_(GoalSelector.java:120)
        at TRANSFORMER/minecraft@1.20.1/net.minecraft.world.entity.ai.goal.GoalSelector.m_25373_(GoalSelector.java:111)
        at TRANSFORMER/minecraft@1.20.1/net.minecraft.world.entity.Mob.m_6140_(Mob.java:760)
        at TRANSFORMER/minecraft@1.20.1/net.minecraft.world.entity.LivingEntity.m_8107_(LivingEntity.java:2548)
        at TRANSFORMER/minecraft@1.20.1/net.minecraft.world.entity.Mob.m_8107_(Mob.java:536)
        at TRANSFORMER/minecraft@1.20.1/net.minecraft.world.entity.monster.Monster.m_8107_(Monster.java:42)
        at TRANSFORMER/minecraft@1.20.1/net.minecraft.world.entity.monster.Zombie.m_8107_(Zombie.java:236)
        at TRANSFORMER/minecraft@1.20.1/net.minecraft.world.entity.LivingEntity.m_8119_(LivingEntity.java:2298)
        at TRANSFORMER/minecraft@1.20.1/net.minecraft.world.entity.Mob.m_8119_(Mob.java:337)
        at TRANSFORMER/minecraft@1.20.1/net.minecraft.world.entity.monster.Zombie.m_8119_(Zombie.java:210)
        at TRANSFORMER/minecraft@1.20.1/net.minecraft.server.level.ServerLevel.m_8647_(ServerLevel.java:693)
        at TRANSFORMER/minecraft@1.20.1/net.minecraft.world.level.Level.m_46653_(Level.java:479)
        at TRANSFORMER/minecraft@1.20.1/net.minecraft.server.level.ServerLevel.m_184063_(ServerLevel.java:343)
        at TRANSFORMER/minecraft@1.20.1/net.minecraft.world.level.entity.EntityTickList.m_156910_(EntityTickList.java:54)
        at TRANSFORMER/minecraft@1.20.1/net.minecraft.server.level.ServerLevel.m_8793_(ServerLevel.java:323)
        at TRANSFORMER/minecraft@1.20.1/net.minecraft.server.MinecraftServer.m_5703_(MinecraftServer.java:893)
        at TRANSFORMER/minecraft@1.20.1/net.minecraft.server.dedicated.DedicatedServer.m_5703_(DedicatedServer.java:283)
        at TRANSFORMER/minecraft@1.20.1/net.minecraft.server.MinecraftServer.m_5705_(MinecraftServer.java:814)
        at TRANSFORMER/minecraft@1.20.1/net.minecraft.server.MinecraftServer.m_130011_(MinecraftServer.java:661)
        at TRANSFORMER/minecraft@1.20.1/net.minecraft.server.MinecraftServer.m_206580_(MinecraftServer.java:251)
        at java.base/java.lang.Thread.run(Thread.java:833)
```

**Comment by twothe** [comment]
After further testing, I think this is an attempt to despawn a villager that gets transformed into a zombie.

**Comment by ricksouth** [comment]
Thank you for opening the issue and giving the mod a try! This has been fixed in version 5.9 of No Hostiles Around Campfire.

https://curseforge.com/minecraft/mc-mods/no-hostiles-around-campfire/files

**Comment by twothe** [comment]
Thank you for the fix, and a happy new year. :)
